File: app/src/main/java/com/motrixi/datacollection/utils/IPAddressUtils.java
package com.motrixi.datacollection.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.net.wifi.WifiInfo;
import android.net.wifi.WifiManager;

import com.adobe.fre.FREContext;

import java.net.Inet4Address;
import java.net.InetAddress;
import java.net.NetworkInterface;
import java.net.SocketException;
import java.util.Enumeration;

/**
 * author : Jason
 * date   : 2020/12/8 3:09 PM
 * desc   :
 */
public class IPAddressUtils {

    public static String getCurrentIPAddress(FREContext context) {
        try {
            String ipAddress = "";
            NetworkInfo info = ((ConnectivityManager) context.getActivity()
                    .getSystemService(Context.CONNECTIVITY_SERVICE)).getActiveNetworkInfo();

            if (info != null && info.isConnected()) {
                if (info.getType() == ConnectivityManager.TYPE_MOBILE) {

                    ipAddress = currentIpAddress();
                } else if (info.getType() == ConnectivityManager.TYPE_WIFI){
                    WifiManager wifiManager = (WifiManager) context.getActivity().getApplicationContext().getSystemService(Context.WIFI_SERVICE);
                    WifiInfo wifiInfo = wifiManager.getConnectionInfo();
                    ipAddress = intIP2StringIP(wifiInfo.getIpAddress());
                }
            }
            return ipAddress;
        } catch (Exception e) {
            return "";
        }
    }

    public static String getCurrentIPAddress(Context context) {
        try {
            String ipAddress = "";
            NetworkInfo info = ((ConnectivityManager) context
                    .getSystemService(Context.CONNECTIVITY_SERVICE)).getActiveNetworkInfo();

            if (info != null && info.isConnected()) {
                if (info.getType() == ConnectivityManager.TYPE_MOBILE) {

                    ipAddress = currentIpAddress();
                } else if (info.getType() == ConnectivityManager.TYPE_WIFI){
                    WifiManager wifiManager = (WifiManager) context.getApplicationContext().getSystemService(Context.WIFI_SERVICE);
                    WifiInfo wifiInfo = wifiManager.getConnectionInfo();
                    ipAddress = intIP2StringIP(wifiInfo.getIpAddress());
                }
            }
            return ipAddress;
        } catch (Exception e) {
            return "";
        }
    }

    private static String currentIpAddress(){

        try {
            for (Enumeration<NetworkInterface> en = NetworkInterface.getNetworkInterfaces(); en.hasMoreElements(); ) {
                NetworkInterface intf = en.nextElement();
                for (Enumeration<InetAddress> enumIpAddr = intf.getInetAddresses(); enumIpAddr.hasMoreElements(); ) {
                    InetAddress inetAddress = enumIpAddr.nextElement();
                    if (!inetAddress.isLoopbackAddress() && inetAddress instanceof Inet4Address) {
                        return inetAddress.getHostAddress();
                    }
                }
            }
        } catch (SocketException e) {
            e.printStackTrace();
        }
        return null;
    }

    public static String intIP2StringIP(int ip) {
        return (ip & 0xFF) + "." +
                ((ip >> 8) & 0xFF) + "." +
                ((ip >> 16) & 0xFF) + "." +
                (ip >> 24 & 0xFF);
    }

}

File: app/src/main/java/com/motrixi/datacollection/utils/IPAddressUtils.kt
package com.motrixi.datacollection.utils

import android.content.Context
import android.net.ConnectivityManager
import android.net.wifi.WifiManager
import android.util.Log
import java.net.*


/**
 * author : Jason
 * date   : 2020/10/15 10:05 PM
 * desc   :
 */
object IPAddressUtils {

    fun getCurrentIPAddress(context: Context): String? {
        try {
            var ipAddress: String? = null
            val manager =
                context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager

            val mobileNetworkInfo = manager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE)
            val wifiNetworkInfo = manager.getNetworkInfo(ConnectivityManager.TYPE_WIFI)
            if (mobileNetworkInfo != null && mobileNetworkInfo!!.isConnected) {
                ipAddress = currentIpAddress
            } else if (wifiNetworkInfo!!.isConnected) {
                val wifiManager = context.getSystemService(Context.WIFI_SERVICE) as WifiManager
                val wifiInfo = wifiManager.connectionInfo
                val address = wifiInfo.ipAddress
                ipAddress = intToStringIP(address)

    //            ipAddress = getOutNetIP(context)
            }
            return ipAddress
        } catch (e: Exception) {
            return ""
        }
    }

    private val currentIpAddress: String?
        get() {
            try {
                val en = NetworkInterface.getNetworkInterfaces()
                while (en.hasMoreElements()) {
                    val net = en.nextElement()
                    val enumIpAddr = net.inetAddresses
                    while (enumIpAddr.hasMoreElements()) {
                        val inetAddress = enumIpAddr.nextElement()
                        if (!inetAddress.isLoopbackAddress && inetAddress is Inet4Address) {
                            return inetAddress.getHostAddress()
                        }
                    }
                }
            } catch (ex: SocketException) {
                Log.e("local IP", ex.toString())
            }
            return null
        }

    private fun intToStringIP(ip: Int): String {
        val sb = StringBuilder()
        sb.append(ip and 0xFF).append(".")
        sb.append(ip shr 8 and 0xFF).append(".")
        sb.append(ip shr 16 and 0xFF).append(".")
        sb.append(ip shr 24 and 0xFF)

        return sb.toString()
    }

}